### DISCUSSION TITLE
Nerd Font icons stopped appearing

### DISCUSSION THREAD

**Discussion by gwenwindflower** [LazyVim/LazyVim#2821] [Q&A]
Hi, I'm looking for any ideas on how to approach troubleshooting this -- nerd font icons stopped appearing for me yesterday, and I can't figure out what might have changed. I'm _definitely_ using a Nerd Font, I patched it myself with all symbols, and it worked until yesterday. Any ideas on how to approach troubleshooting? Using latest version of kitty terminal.

Thanks for any help!

<img width="1139" alt="Screenshot 2024-03-25 at 6 52 12 AM" src="https://github.com/LazyVim/LazyVim/assets/91998347/16b926aa-edf3-41f9-9e02-bcefac40a8af">
<img width="543" alt="Screenshot 2024-03-25 at 6 55 22 AM" src="https://github.com/LazyVim/LazyVim/assets/91998347/55ae7e66-2124-49c4-a360-48edec7c75d7">

**Comment by gwenwindflower** [comment]
I repatched the font no dice, but I did some testing with a non-patched font and the Symbols Only font installed, and while other symbols were wonky and needed to be symbol mapped, these came back. So this is definitely just a font issue, will troubleshoot. Closing as this is not LazyVim-related I don't think.

**Comment by rrpolanco** [comment]
I just updated to the latest LazyVIm (`10.18.0`) and I'm seeing the same issue:
<img width="1945" alt="image" src="https://github.com/LazyVim/LazyVim/assets/6497491/4fec09b8-43e1-4e1d-9f37-5fbc40ca8840">

I have Nerdfont 3.0 installed and used as the fallback font. Symbols were working prior to this update. I'm investigating.

**Reply by gwenwindflower** [reply]
are you using Kitty? i found the fix thread last night if so.

**Reply by rrpolanco** [reply]
I'm using Kitty terminal. Are there any additional symbol mappings required?

**Reply by gwenwindflower** [reply]
https://github.com/kovidgoyal/kitty/issues/7249

**Reply by gwenwindflower** [reply]
TLDR if you install the nightly build the fix is in already

**Reply by gwenwindflower** [reply]
also learned a fun thing from that thread that `ga` will output the unicode code points of whatever your cursor is over, so if you find additional mappings missing that's how you can find what you need to map

**Reply by rrpolanco** [reply]
Thanks @gwenwindflower! I'll wait for an official Kitty release. I didn't know about `ga` either!

**Reply by gwenwindflower** [reply]
cool cool! in the meantime if you `ga` those missing arrows in whichkey, just be sure to map them to Monaco or Menlo (or whatever your system's default mono font is) not Symbols Nerd Font. For reasons that are a litle beyond my patience for close reading. It does work though can confirm before I installed nightly.

**Comment by dpetka2001** [comment]
Is this a OS specific problem? Or font specific? I also use Kitty on Linux Mint 21.3 and haven't observed this problem with version 0.33.1. I use non-patched fonts with only `symbol_map` mapped to `Symbols Nerd Font`.

**Reply by gwenwindflower** [reply]
i don't think they could pin it down fully @dpetka2001, i skimmed parts of the thread so i could be wrong -- but i did see it was working on kovid's mac, but not somebody else's, so not OS specific, but i didn't read closely enough to fully understand the root issue, once i saw it was fixed i just trusted kovid understood it and installed nightly. it's something about the fallback fonts available on the system and dingbats messing things up i believe.

**Reply by dpetka2001** [reply]
The other guy mentioned that he has the same version of OS. Do you also observe this problem on Mac? Or do you have different OS?

**Reply by dpetka2001** [reply]
In his [patch-commit](https://github.com/kovidgoyal/kitty/commit/4878b7cfd3b594b863280298bb972cd3026bb385) he mentions 
> if the original fallback font is Zapf Dingbats we get .AppleSystemUIFontMonospaced as the new fallback

So, that leads me to believe that it's Mac specific. That must be why I haven't observed such behavior at all. Very useful thread you linked @gwenwindflower and nice find (even if it was from the thread) about `ga`. Didn't know about that, so thanks for linking that issue :smile:

**Reply by gwenwindflower** [reply]
oh yea sorry @dpetka2001 , i said that wrong -- i meant not OS-specific as in the OS is seemingly not the root issue and it's not universal on all Macs even those running the same OS, but i believe you're absolutely right that it's just a Mac thing, just a specific subset of Macs.
